pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/TableDataManagerProvider.java

```diff
@@ -45,6 +46,11 @@ public static void init(InstanceDataManagerConfig instanceDataManagerConfig) {
     if (maxParallelBuilds > 0) {
       _segmentBuildSemaphore = new Semaphore(maxParallelBuilds, true);
     }
+    SegmentBuildTimeLeaseExtender.initExecutor();
```

Comment:
Suggest moving the `initExecutor()` and `shutdownExecutor()` to `HelixInstanceDataManager`.